odpf/siren/v1/siren.proto

```diff
@@ -0,0 +1,343 @@
+            post: "/templates/{name}/render",
+            body: "*"
+        };
+    }
+}
+
+message PingRequest {}
+
+message PingResponse {
+    string message = 1;
+}
+
+message ListAlertHistoryRequest {
+    string resource = 1;  
+    uint32 start_time = 2;
+    uint32 end_time = 3;
+}
+
+message ListAlertHistoryResponse {
+    repeated AlertHistory alerts = 1;
+}
+
+message AlertHistory {
+    uint64 id = 1;
+    string name = 2;
+    string template_id = 3;
+    string metric_name = 4;
+    string metric_value = 5;
+    string level = 6;
+    string created_at = 7;
+    string updated_at = 8;
+}
+
+message Annotations {
+    string metric_name = 1;
+    string metric_value = 2;
+    string resource = 3;
+    string template = 4;
+}
+
+message Labels {
+    string severity = 1;
+}
+
+message Alert {
+    Annotations annotations = 1;
+    Labels labels = 2;
+    string status = 3;
+}
+
+message CreateAlertHistoryRequest {
+    repeated Alert alerts = 1;
+}
+
+message CreateAlertHistoryResponse {
+    repeated AlertHistory alerts = 1;
+}
+
+message Workspace {
+    string id = 1;
+    string name = 2;
+}
+
+message ListWorkspaceChannelsRequest {
+    string workspace_name = 1 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+}
+
+message ListWorkspaceChannelsResponse {
+    repeated Workspace data = 1;
+}
+
+message ExchangeCodeRequest {
+    string code = 1 [(validate.rules).string.pattern = "^[A-Za-z0-9._-]+$"];
+    string workspace = 2 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+}
+
+message ExchangeCodeResponse {
+    bool ok = 1;
+}
+
+message GetAlertCredentialsRequest{
+    string team_name = 1 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+}
+
+message Critical {
+    string channel = 1 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+}
+
+message Warning {
+    string channel = 1 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+}
+
+message SlackConfig {
+    Critical critical = 1 [(validate.rules).message.required = true];
+    Warning warning = 2 [(validate.rules).message.required = true];
+}
+
+message GetAlertCredentialsResponse {
+    string entity = 1;
+    string team_name = 2;
+    string pagerduty_credentials = 3;
+    SlackConfig slack_config = 4;
+}
+
+message UpdateAlertCredentialsRequest {
+    string entity = 1 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+    string team_name = 2;
+    string pagerduty_credentials = 3 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+    SlackConfig slack_config = 4 [(validate.rules).message.required = true];
+}
+message UpdateAlertCredentialsResponse{}
+
+
+message SendSlackNotificationRequest{
+    string provider = 1;
+    string entity = 2 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+    string message = 3;
+    string receiver_name = 4 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+    string receiver_type = 5 [(validate.rules).string = {in: ["channel", "user"]}];;
```

Comment:
here also, can we make reciever_type as enum? wdyt?

Reply:
Not doing this too, First, we will need to find a solution for Enum proto to pass `case insensitive` string.